runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -195,32 +203,40 @@ private static Object handleReadDataCorruption(CorfuPayloadMsg<ReadResponse> msg
      */
     public CompletableFuture<Boolean> write(long address, Set<UUID> streams, long rank,
                                             ByteBuf buffer, Map<UUID, Long> backpointerMap) {
+        Timer.Context context = getTimerContext("writeByteBuf");
         WriteRequest wr = new WriteRequest(WriteMode.NORMAL, null, buffer);
         wr.setStreams(streams);
         wr.setRank(rank);
         wr.setBackpointerMap(backpointerMap);
         wr.setGlobalAddress(address);
-        return router.sendMessageAndGetCompletable(CorfuMsgType.WRITE.payloadMsg(wr));
+        CompletableFuture<Boolean> cf = router.sendMessageAndGetCompletable(CorfuMsgType.WRITE.payloadMsg(wr));
+        return cf.thenApply(x -> { context.stop(); return x; });
     }
 
     public CompletableFuture<Boolean> writeStream(long address, Map<UUID, Long> streamAddresses,
                                                   Object object) {
+        Timer.Context context = getTimerContext("writeStreamObject");
```

Comment:
do we not need counters for these function calls ? It can tell us if there is any correlation between time taken per invocation and number of invocations.